src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/jax_rs/ResourceMethodMultipleEntityParamsQuickFix.java

```diff
@@ -155,4 +148,19 @@ private boolean isEntityParam(PsiParameter param) {
         }
         return true;
     }
+
+    public CodeAction createCodeAction(JavaCodeActionContext context, Diagnostic diagnostic,
+                                              String quickFixMessage, String participantId, Integer entityParamIndex) {
+        ExtendedCodeAction codeAction = new ExtendedCodeAction(quickFixMessage);
+        codeAction.setRelevance(0);
+        codeAction.setDiagnostics(Collections.singletonList(diagnostic));
+        codeAction.setKind(CodeActionKind.QuickFix);
+        Map<String, Object> extendedData = new HashMap<>();
+        extendedData.put(ENTITY_PARAM_INDEX_KEY, entityParamIndex);
+        codeAction.setData(new CodeActionResolveData(context.getUri(), participantId,
+                context.getParams().getRange(), extendedData,
+                context.getParams().isResourceOperationSupported(),
+                context.getParams().isCommandConfigurationUpdateSupported()));
+        return codeAction;
+    }
```

Comment:
This method is mostly a copy of `JDTUtils.createCodeAction()` with the main difference being the addition of the HashMap. Seems like a more generalized version belongs in `JDTUtils` and should be called from here instead of duplicating the code again.

Reply:
Modification done